### PR TITLE
Addressed memory leak in S3/S4 plotting, new defaults for NIRSpec S3 ecf

### PR DESCRIPTION
**Pull request by erinmmay**
resolves issue #142. Updated S3 ecf file for NIRSpec is a good starting point for getting decent light curves.

### CODE REVIEW

File: eureka/S3_data_reduction/plots_s3.py [hunk 1]
```diff
@@ -116,15 +116,15 @@ def optimal_spectrum(data, meta, n):
     plt.suptitle(f'1D Spectrum - Integration {intstart + n}')
     plt.semilogy(range(subnx), stdspec[n], '-', color='C1', label='Standard Spec')
     # plt.errorbar(range(subnx), stdspec[n], yerr=np.sqrt(stdvar[n]), fmt='-', color='C1', ecolor='C0', label='Std Spec')
-    plt.errorbar(range(subnx), optspec[n], opterr[n], fmt='-', color='C2', ecolor='C2', label='Optimal Spec')
+    # plt.errorbar(range(subnx), optspec[n], opterr[n], fmt='-', color='C2', ecolor='C2', label='Optimal Spec')
```
Comment (kevin218) [inline eureka/S3_data_reduction/plots_s3.py:119]: Why remove optimal spectrum from figure?

File: eureka/S4_generate_lightcurves/plots_s4.py [hunk 2]
```diff
@@ -26,17 +26,17 @@ def binned_lightcurve(meta, time, i):
...
     plt.ylabel('Normalized Flux')
     plt.xlabel(f'Time [{meta.time_units} - {time_modifier}]')
 
     plt.subplots_adjust(left=0.10, right=0.95, bottom=0.10, top=0.90, hspace=0.20, wspace=0.3)
     plt.savefig(meta.outputdir + 'figs/Fig' + str(4300 + i) + '-' + meta.eventlabel + '-1D_LC.png')
-    if meta.hide_plots:
-        plt.close()
-    else:
+    if not meta.hide_plots:
+    #     plt.close()
```
Comment (kevin218) [inline eureka/S4_generate_lightcurves/plots_s4.py:38]: Remove commented out lines

File: eureka/S3_data_reduction/plots_s3.py (hunk 1 above)
Comment (erinmmay) [reply eureka/S3_data_reduction/plots_s3.py:119]: oops, I'll uncomment that. This was to get around a weird bug where some values are treated as strings. There's a different issue open related to this.

File: eureka/S4_generate_lightcurves/plots_s4.py (hunk 2 above)
Comment (erinmmay) [reply eureka/S4_generate_lightcurves/plots_s4.py]: done

File: eureka/S3_data_reduction/plots_s3.py [hunk 3]
```diff
@@ -44,9 +44,9 @@ def lc_nodriftcorr(meta, wave_1d, optspec):
     plt.colorbar(label='Normalized Flux')
     plt.tight_layout()
     plt.savefig(meta.outputdir + 'figs/fig3101-2D_LC.png')
-    if meta.hide_plots:
-        plt.close()
-    else:
+    if not meta.hide_plots:
+    #     plt.close()
```
[the remark this answers is not in the document]
Comment (erinmmay) [reply eureka/S3_data_reduction/plots_s3.py]: done

**Review by kevin218 — APPROVED** [review]

**Pull request merged**